ledger/src/blockstore_meta.rs

```diff
@@ -406,6 +406,35 @@ impl ErasureMeta {
     }
 }
 
+#[allow(dead_code)]
+impl MerkleRootMeta {
+    pub(crate) fn from_shred(shred: &Shred) -> Self {
+        Self {
+            // An error here after the shred has already sigverified
+            // can only indicate that the leader is sending
+            // legacy or malformed shreds. We should still store
+            // `None` for those cases in blockstore, as a later
+            // shred that contains a proper merkle root would constitute
+            // a valid duplicate shred proof.
```

Comment:
duplicate "block" proof, no?
we might not actually observe any duplicate shreds.

Reply:
not sure about the terminology. We're still calling them duplicate shred proofs in gossip even though it has evolved beyond shreds with the same index. If you prefer I can do a separate refactor PR afterwards renaming all instances of "duplicate shred proof" to "duplicate block proof"?

Reply:
never mind then,
we can revisit once the code is shipped and stable.